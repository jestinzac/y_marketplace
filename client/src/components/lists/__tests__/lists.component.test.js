import { render, screen } from "@testing-library/react";

import Lists from "../lists.component";

describe("List", () => {
  it("should render the list container with respective items", () => {
    const mockData = {
      id: 5,
      username: "mario.bros",
      jobs: [{
        id: 2,
        tier: "Plumbers", 
        headline: "Leaky tap needs fixing", 
        description: "The tap in my kitchen is dripping all of the time, please fix it.", 
        date_added: "2020-11-05T15:47:54Z", 
        area: "Brooklyn, NYC"
      }]
    }
    const { asFragment } = render(
      <Lists {...mockData} />
    );

    expect(screen.getByText("Leaky tap needs fixing")).toBeInTheDocument();
    expect(screen.getByTestId("list-container")).toHaveAttribute("data-username", "mario.bros");
    expect(screen.getByTestId("calender-svg-icon")).toBeInTheDocument();
    expect(screen.getByTestId("map-pin-svg-icon")).toBeInTheDocument();
    expect(screen.getByTestId("formatted-date").textContent).toBe("11/5/2020, 15:47");
    expect(screen.getByTestId("list-item")).toHaveAttribute("id", "2");
    expect(asFragment()).toMatchSnapshot();
  });

  it("should show info box with respective message if its not error but not able to find in the list items", () => {
    const mockData = {
      id: 5,
      username: "mario.bros",
      jobs: []
    }
    const { asFragment } = render(
      <Lists {...mockData} hasError={false} />
    );

    expect(screen.getByTestId("list-container")).toHaveAttribute("data-username", "mario.bros");
    expect(screen.queryByTestId("calender-svg-icon")).not.toBeInTheDocument();
    expect(screen.queryByTestId("map-pin-svg-icon")).not.toBeInTheDocument();

    expect(screen.getByTestId("info-box")).toBeInTheDocument();
    expect(screen.getByText("No jobs available at the moment.")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  })

});
