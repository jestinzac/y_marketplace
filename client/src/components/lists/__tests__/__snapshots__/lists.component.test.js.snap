// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render the list container with respective items 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen flex flex-col list-items"
    data-testid="list-container"
    data-username="mario.bros"
  >
    <article
      class="w-auto p-6 m-5 bg-white rounded-md overflow-hidden shadow-lg hover:shadow-xl list-item"
      data-testid="list-item"
      id="2"
    >
      <span
        class="rounded-full bg-purple-200 py-1 px-4 text-xs font-bold text-purple-600 tag"
      >
        Plumbers
      </span>
      <h2
        class="text-3xl py-4 font-bold text-blue-800 description headline"
      >
        Leaky tap needs fixing
      </h2>
      <p
        class="text-xl py-3 pb-20 font-medium text-gray-900"
      >
        The tap in my kitchen is dripping all of the time, please fix it.
      </p>
      <div
        class="py-4 flex justify-between items-center bottom-info"
      >
        <div
          class="flex gap-2"
        >
          <svg
            class="w-6 h-6"
            data-testid="calender-svg-icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.75 3v2.25M17.25 3v2.25M3 18.75V7.5a2.25 2.25 0 012.25-2.25h13.5A2.25 2.25 0 0121 7.5v11.25m-18 0A2.25 2.25 0 005.25 21h13.5A2.25 2.25 0 0021 18.75m-18 0v-7.5A2.25 2.25 0 015.25 9h13.5A2.25 2.25 0 0121 11.25v7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <span
            class="date content-end"
            data-testid="formatted-date"
          >
            11/5/2020, 15:47
          </span>
        </div>
        <div
          class="flex gap-2"
        >
          <svg
            class="w-6 h-6"
            data-testid="map-pin-svg-icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 10.5a3 3 0 11-6 0 3 3 0 016 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              d="M19.5 10.5c0 7.142-7.5 11.25-7.5 11.25S4.5 17.642 4.5 10.5a7.5 7.5 0 1115 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <span
            class="location"
          >
            Brooklyn, NYC
          </span>
        </div>
      </div>
    </article>
  </div>
</DocumentFragment>
`;

exports[`List should show info box with respective message if its not error but not able to find in the list items 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen flex flex-col list-items"
    data-testid="list-container"
    data-username="mario.bros"
  >
    <div
      class="block text-sm text-left h-12 flex items-center p-4 rounded-sm border text-indigo-600 bg-indigo-200 border-indigo-400"
      data-testid="info-box"
      data-type="info"
      role="alert"
    >
      No jobs available at the moment.
    </div>
  </div>
</DocumentFragment>
`;
